### PR TITLE
Optimize `_approve` and add test

### PR DESCRIPTION
**Pull request by Vectorized**
Saves 20 gas. 

Push back test coverage back up to 100%.

### CODE REVIEW

**Comment by cygaar** [comment]
Can you wait for code reviews before merging in?

**Comment by Vectorized** [comment]
Ok

**Review by cygaar — APPROVED** [review]

**Pull request merged**
